### PR TITLE
Rename `install_buildbuddy_dependencies` => `install_go_mod_dependencies`

### PR DESCRIPTION
**Pull request by siggisim**
This change:
 1) Allows us to have different functions for different language ecosystems (i.e. install_package_json_dependencies, etc)
2) Makes it easier for automated tooling to generate this function name (i.e `install_%s_dependencies` where %s is the name of the file where the dependencies came from). 

I have a separate change do to the same in the internal repo.

### CODE REVIEW

File: deps.bzl [hunk 1]
```diff
@@ -1,8 +1,8 @@
 load("@bazel_gazelle//:deps.bzl", "go_repository")
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive", "http_file")
 
-# bazelisk run //:gazelle -- update-repos -from_file=go.mod -to_macro=deps.bzl%install_buildbuddy_dependencies
-def install_buildbuddy_dependencies(workspace_name = "buildbuddy"):
+# bazelisk run //:gazelle -- update-repos -from_file=go.mod -to_macro=deps.bzl%install_go_mod_dependencies
+def install_go_mod_dependencies(workspace_name = "buildbuddy"):
```
Comment (tylerwilliams) [inline deps.bzl:5]: I think there are some http_archives and http_files that we might wanna put in a go_static_dependencies target or something if this is gonna be go_mod only

File: WORKSPACE [hunk 2]
```diff
@@ -193,6 +193,9 @@ http_archive(
 
 load("@com_google_googletest//:googletest_deps.bzl", "googletest_deps")
 
+# gazelle:repository_macro deps.bzl%install_go_mod_dependencies
+install_go_mod_dependencies()
```
Comment (vadimberezniker) [inline WORKSPACE:197]: Why is this being called twice?

**Review by vadimberezniker — CHANGES_REQUESTED** [review]

File: deps.bzl (hunk 1 above)
Comment (siggisim) [reply deps.bzl:5]: Pulled those out into their own function - there's no harm in having them together, but this makes it clearer which ones are generated and which ones are manual.

File: WORKSPACE (hunk 2 above)
Comment (siggisim) [reply WORKSPACE]: Removed

**Review by tylerwilliams — APPROVED** [review]

**Review by vadimberezniker — APPROVED** [review]

**Pull request merged**